test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresUnreferencedCodeCapability.cs

```diff
@@ -15,6 +15,8 @@ namespace Mono.Linker.Tests.Cases.RequiresCapability
 	[SkipKeptItemsValidation]
 	public class RequiresUnreferencedCodeCapability
 	{
+		[ExpectedWarning ("IL2026", "Calling 'Mono.Linker.Tests.Cases.RequiresCapability.RequiresUnreferencedCodeCapability.BaseType.VirtualMethodRequiresUnreferencedCode()' " +
+			"which has `RequiresUnreferencedCodeAttribute` can break functionality when trimming application code. Message for --VirtualMethodRequiresUnreferencedCode--.")]
```

Comment:
This is really weird - the place where the warning is reported....

Reply:
I agree - although I'm not sure what to do in cases like this, where a method is kept because of a `DynamicallyAccessedMember` attribute. The method is not being called, we are just marking it in case the user later calls it, so I think maybe this should not warn.

Reply:
You're right that we preserve it because it may be needed later - but that's kind of the point. Linker can't figure out when exactly it's going to be used, so instead we have data flow annotations and will "detect" the usage somewhere else. Since we can't tell where it's actually being used, there's no point trying to report it there. Instead we should report it where we know about it - it's not 100% but it's pretty close.

Not warning is not an option - if the method is going to be used for real (linker can't tell), that would break the promise and potentially the app. So we have to play it safe and warn.

I think the linker should warn at the place where it detects the need to preserve the method - in cases of DynamicallyAccessedMember, it's the same place where we would issue warnings about problems with the DynamicallyAccessedMember attribute itself (for example if the type being used is not statically known).